ports/esp32/main.c

```diff
@@ -99,9 +99,9 @@ void mp_task(void *pvParameter) {
     #if MICROPY_PY_THREAD
     mp_thread_init(pxTaskGetStackStart(NULL), MICROPY_TASK_STACK_SIZE / sizeof(uintptr_t));
     #endif
-    #if CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG_ENABLED
+    #if MICROPY_HW_ESP_USB_SERIAL_JTAG
     usb_serial_jtag_init();
-    #elif CONFIG_USB_OTG_SUPPORTED
+    #elif MICROPY_HW_USB_CDC
```

Comment:
I like this, using `MICROPY_HW_xxx` config options exclusively makes it easier to follow.

But, it seems this will clash with #15108, which uses `MICROPY_HW_ENABLE_USBDEV` here instead of `MICROPY_HW_USB_CDC`.  Maybe the former is better because you want USB to be initialised if any USB mode is used (eg just MSC)?

I guess a lot of this PR will clash with #15108, eg the changes to `mphalport.c`.  Don't want to have our efforts here wasted if #15108 will need to change things anyway.

Reply:
Yes, agree that eventually we want to split this into `MICROPY_HW_ENABLE_USBDEV` and `MICROPY_HW_USB_CDC` the same as the rp2 port. This did cross my mind when creating this, but I figured at the moment these two macros would be equivalent - so leave splitting them out for when introducing the functionality that needs to differentiate them. I can see the case for adding it now, though - a little less churn.

@andrewleech Is doing this one way or the other way going to be easier for you?

Reply:
Looked at #15108 again and can see the point, so I've pushed commit e1fd26d82 that adds `MICROPY_HW_ENABLE_USBDEV`. It's a bit odd because if you build with `#define MICROPY_HW_ENABLE_USBDEV (1)` and `#define MICROPY_HW_USB_CDC (0)` then the port enumerates as a USB-CDC device (as the descriptor is configured in ESP-IDF) but the port doesn't do anything.

It might be better if we drop that commit from here, and @andrewleech adds it to his branch the next time he rebases #15108. Not sure.

Reply:
Let's make the change to use `MICROPY_HW_ENABLE_USBDEV` in #15108 instead.  That keeps this PR simple.